src/main/java/com/google/sps/servlets/ProfileRoutes.java

```diff
@@ -30,13 +30,23 @@
 @SuppressWarnings("serial")
 @WebServlet("/user-routes")
 public class ProfileRoutes extends HttpServlet {
+  static class RouteWithType {
+    Route route;
+    Long type;
```

Comment:
`type` is not a very descriptive name. It is not clear what it means. Perhaps use `displayMode` or `userAccess`, or `accessMode` that makes it clear what the field will be used for.

Also, it would be better to use an enum, which makes clear what each number means, rather than just a number.